specs/accordion.md

```diff
@@ -0,0 +1,194 @@
+- [Lightning Design](https://www.lightningdesignsystem.com/components/accordion/)
+- [Fluent UI](https://fluentsite.z22.web.core.windows.net/components/accordion/accessibility)
+- [Primer](https://primer.style/components/Details)
+- [Bootstrap](https://getbootstrap.com/docs/4.3/components/collapse/)
+
+---
+
+### API
+
+**Accordion**
+*Component Name*
+- `fast-accordion`
+
+*Attributes:*
+- `expand` - enum
+  - single
+  - multi - default
+
+*Events*
+- `change: CustomEvent`
+  - no custom data
+  - bubbles
+
+**Accordion Item**
+*Component names:*
+- `fast-accordion-item`
+
+*Attributes:*
+- `expanded` - boolean
+- `id` - string
+
+*Parts:*
+- panel
+
+*Slot Names*
+- header
+
+**Accordion Header**
+*Component names:*
+- `fast-accordion-header`
+
+*Attributes:*
+- `expanded` - boolean
+- `id` - string
+- `headingtag` - enum
+  - h1
+  - h2
+  - h3 - default
+  - h4
+  - h5
+  - h6
+
+*Parts:*
+- button
+
+*Slot Names*
+- expanded-glyph
+- collapsed-glyph
+
+### Anatomy and Appearance
+
+```HTML
+<!-- shadow root -->
+<div>
+    <h3>
+        <button
+            type="button"
+            part="button"
+            aria-expanded="true"
+            class="button"
+            aria-controls="panel1"
+            id="accordion1"
+        >
+            Panel one
+            <slot name="expanded-glyph" part="expanded-glyph"></slot>
+            <slot name="collapsed-glyph" part="collapsed-glyph"></slot>
+       </button>
+    </h3>
+    <div
+        id="panel1"
+        role="region"
+        aria-labelledby="accordion1"
+    >
+        Panel one content
+    </div>
+    <h3>
+        <button
+            type="button"
+            aria-expanded="false"
+            class="button"
+            aria-controls="panel2"
+            id="accordion2"
+        >
+            Panel one
+            <slot name="expanded-glyph" part="expanded-glyph"></slot>
+            <slot name="collapsed-glyph" part="collapsed-glyph"></slot>
+       </button>
+    </h3>
+    <div
+        id="panel2"
+        part="panel"
+        role="region"
+        aria-labelledby="accordion2"
+        class="panel"
+    >
+        Panel one content
+    </div>
+    <h3>
+        <button
+            type="button"
+            aria-expanded="true"
+            class="button"
+            aria-controls="panel3"
+            id="accordion3"
+        >
+            Panel one
+            <slot name="expanded-glyph" part="expanded-glyph"></slot>
+            <slot name="collapsed-glyph" part="collapsed-glyph"></slot>
+       </button>
+    </h3>
+    <div
+        id="panel3"
+        part="panel"
+        role="region"
+        aria-labelledby="accordion3"
+        class="panel"
+    >
+        Panel one content
+    </div>
+</div>
+
+<!-- end shadow root -->
+```
+
+---
+
+## Implementation
+
+```HTML
+<fast-accordion>
+    <fast-accordion-item>
+        <fast-accordion-header headingtag="h2">Header One</fast-accordion-header>
```

Comment:
Was there a reason this couldn't actually be an `h2` tag with a `heading` slot?

Reply:
because we need to know if its expanded or not to display the proper glyph

Reply:
![image](https://user-images.githubusercontent.com/37851214/80418121-bf53e700-888b-11ea-83e4-a06cdc6c6813.png)

Reply:
Does the glyph need to be inside of the header tag or could it be beside it?

Reply:
I think it should be inside... That's how the w3c example shows it.

https://w3c.github.io/aria-practices/examples/accordion/accordion.html

Reply:
Ok, that makes sense.

Note that making the tag swaps right now may be a bit difficult with the templating engine today. I've got features planned that will make that easier but doubtful they'll get in soon based on my current workload.

Reply:
We can probably circumvent tag-switching by using `role="heading"` and `aria-lavel="1|2|3|4|5|6"`

Reply:
I'm still not entirely convinced we need a separate `header` item:

```
<fast-accordion-item>
  <span slot="heading">I'm the heading</slot>
</fast-accordion-item>

<fast-accordion-item heading-level="3">
  <span slot="heading">I'm the heading</slot>
</fast-accordion-item>

<fast-accordion-item heading-level="3">
  <span slot="heading">I'm the heading</slot>
  <span slot="expanded-indicator">Some glyph</span>
  <span slot="collapsed-indicator">Some other glyph</span>
</fast-accordion-item>
```

Reply:
If we do have a separate header item, we definitely want to use the role and level aria attrs instead of switching the tag. That will be more performant even if I had the new template features in place.

Reply:
The only down side I see here is the the `<button/>` needs to be inside the `<h1/>` , `<h2/>`, ... etc. Maybe the template could look something like:

```HTML
    <div role="heading" aria-level=${x => x.headinglevel}>
        <button
            type="button"
            part="button"
            class="button"
            aria-expanded=${x => x.expanded}
            aria-controls=`${x => x.id}-panel`
            id=${x => x.id}
        >
            <slot name="heading"></slot>
            <slot name="expanded-glyph" part="expanded-glyph"></slot>
            <slot name="collapsed-glyph" part="collapsed-glyph"></slot>
       </button>
    </div>
    <div
        id=`${x => x.id}-panel`
        role="region"
        aria-labelledby=${x => x.id}
    >
        <slot></slot>
    </div>
```